src/shogun/multiclass/LaRank.cpp

```diff
@@ -653,16 +654,24 @@ bool CLaRank::train_machine(CFeatures* data)
 			}
 		}
 
-		SG_DEBUG("End of iteration %d\n", n_it++)
+		SG_DEBUG("End of iteration %d\n", n_it)
 		SG_DEBUG("Train error (online): %f%%\n", (tr_err / nb_train) * 100)
 		gap = computeGap ();
 		SG_ABS_PROGRESS(gap, -CMath::log10(gap), -CMath::log10(DBL_MAX), -CMath::log10(get_C()), 6)
 
 		if (!batch_mode)        // skip stopping criteria if online mode
 			gap = 0;
+                n_it++;
 	}
 	SG_DONE()
 
+        if (max_iteration > 0 && n_it >= max_iteration && gap > get_C())
+        {
+            SG_WARNING("LaRank did not converge after %d iterations\n", max_iteration)
+            SG_WARNING("Increase max iteration by calling CLaRank::set_max_iteration()\n")
```

Comment:
No need for stating the method here. Just say "LaRank did not converge after the maximum number of %d iterations\n"

It is pretty clear that increasing this number will then help